### PR TITLE
feat: add EnableArtifactStreaming to NBC and add helper code

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This adds the `EnableArtifactStreaming` field to the NodeBootstrappingConfig so it can be enabled on AKS nodes during provisioning. Additional code for scaffolding is also part of this. With just this PR, enabling it should be a no-op.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 1]
```diff
@@ -150,6 +150,7 @@ KUBENET_TEMPLATE="{{GetKubenetTemplate}}"
 CONTAINERD_CONFIG_CONTENT="{{GetContainerdConfigContent}}"
 CONTAINERD_CONFIG_NO_GPU_CONTENT="{{GetContainerdConfigNoGPUContent}}"
 IS_KATA="{{IsKata}}"
+ARTIFACT_STREAMING_ENABLED="{{ArtifactStreamingEnabled}}"
```
Comment (anujmaheshwari1) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:153]: if we follow the other fields, this should have a prefix like "Is", "Has" or "Get".
For ArtifactStreaming, maybe "IsArtifactStreamingEnabled" is appropriate?

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -953,6 +953,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"GetPrivateEgressProxyAddress": func() string {
 			return config.ContainerService.Properties.SecurityProfile.GetProxyAddress()
 		},
+		"ArtifactStreamingEnabled": func() bool {
```
Comment (anujmaheshwari1) [inline pkg/agent/baker.go:956]: we should add a UT in baker_test.go?

File: parts/linux/cloud-init/artifacts/cse_cmd.sh (hunk 1 above)
Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:153]: There is a mix of fields, like `FIPSEnabled` and others which have such a prefix. I think either one is fine for NBC. What do you think?

File: pkg/agent/baker.go (hunk 2 above)
Comment (ganeshkumar5699) [reply pkg/agent/baker.go:956]: Will be adding it after the behavior change in the next PR. If `ArtifactStreamingEnabled` is true, then it should have a particular containerd config. If not true, then the standard one.

File: parts/linux/cloud-init/artifacts/cse_cmd.sh (hunk 1 above)
Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:153]: I personally like `ENABLE_ARTIFACT_STREAMING` a bit better, along with `EnableArtifactStreaming`, which is what you already have :)

**Review by cameronmeissner — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/cse_cmd.sh (hunk 1 above)
Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh]: Discussed and we want to keep the state, so switched to a combination of both.

**Pull request merged**